src/Native/System.Native/pal_networking.cpp

```diff
@@ -30,6 +30,24 @@
 #include <unistd.h>
 #include <vector>
 
+#if !HAVE_KEVENT64_S
+// On platforms, such as FreeBSD, where kevent64_s
+// is not available, fallback to custom definition
+// with wrapper functions.
+typedef struct kevent kevent64_s;
+
+static void EV_SET64(kevent64_s *kev, uint64_t ident, int16_t filter, uint16_t flags, uint32_t fflags, int64_t data, uint64_t udata, uint64_t ext_0, uint64_t ext_1)
```

Comment:
Nit: this could just be a macro definition: `#define EV_SET64(kev, ident, filter, flags, fflags, data, udata, ext_0, ext_1) EV_SET((kev), (ident), (filter), (flags), (fflags), (data), (udata))`